record_imu_cam/plugin.cpp

```diff
@@ -0,0 +1,99 @@
+#include "common/phonebook.hpp"
+#include "common/plugin.hpp"
+#include "common/switchboard.hpp"
+#include "common/data_format.hpp"
+
+#include <fstream>
+#include <filesystem>
+#include <opencv2/core/mat.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/imgproc.hpp>
+
+using namespace ILLIXR;
+
+class record_imu_cam : public plugin {
+public:
+	record_imu_cam(std::string name_, phonebook* pb_)
+		: plugin{name_, pb_}
+		, sb{pb->lookup_impl<switchboard>()}
+	{
+		sb->schedule<imu_cam_type>(id, "imu_cam", [this](switchboard::ptr<const imu_cam_type> datum, std::size_t){
+			this->dump_data(datum);
+		});
+
+		std::string record_data = get_record_data_path();
+		//check folder exist, if exist delete it'
+		system(("rm  -r " + record_data).c_str());
+
+		std::string imu_file = record_data + "imu0/" + "data.csv";
+		system(("mkdir -p " + record_data + "imu0/").c_str());
+		imu_wt_file.open(imu_file, std::ofstream::out);
+		imu_wt_file << "#timestamp [ns],w_RS_S_x [rad s^-1],w_RS_S_y [rad s^-1],w_RS_S_z [rad s^-1],a_RS_S_x [m s^-2],a_RS_S_y [m s^-2],a_RS_S_z [m s^-2]" << std::endl;
```

Comment:
What's `_RS_S_`?

Reply:
That line is the very first line in the EuRoC's `imu0/data.csv`. I can remove the `_RS_S_` as I'm also not sure what it stands for.